pages/room/[id].js

```diff
@@ -32,11 +34,35 @@ export default function Room(props) {
   const [currPlayer, setCurrPlayer] = useState({
     id: null,
     name: null,
-    card: selectedCard,
+    card: null,
+    room: props.roomID,
   })
 
   const [isOwner, setIsOwner] = useState(false)
 
+  const [updatedUser] = useMutation(UPDATE_USER)
+
+  let onCardClickHandler = async (e, card) => {
+    setSelectedCard(card)
```

Comment:
Not directly related to the PR but I'd consider removing setSelectedCard and replace it with matching on users for current card selected. Also, the currPlayer logic to just point to your own user in users